trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestLoggerController.java

```diff
@@ -107,12 +184,30 @@ public RequestLoggingSession currentRequestSession(UUID requestId)
         return requireNonNull(sessions.get(requestId), "No RequestLoggingSession for requestId: " + requestId);
     }
 
+    public List<SaveEntry> savedEntries(boolean startFromHead, Predicate<SaveEntry> predicate)
+    {
+        return saveQueue
+                .stream()
+                .filter(predicate)
+                .sorted(startFromHead ? COMPARATOR : REVERSED_COMPARATOR)
+                .collect(toImmutableList());
```

Comment:
I see this is used in #120 , but the usecase is to stream the list and collect it again.

Could this function return a `Stream<SaveEntry>` instead?

Reply:
#120 needs to be updated. That re-stream/collect is unnecessary.